erdos/op.py

```diff
@@ -32,11 +33,16 @@ def __init__(self, name, checkpoint_enable=False, checkpoint_freq=None):
         self.progress_tracker = None
         self.framework = None
         self._stream_to_high_watermark = {}
+        self._stream_ignore_watermarks = set()  # input streams that do not send watermarks
+
+        # Checkpoint variables
         self._checkpoint_enable = checkpoint_enable
         self._checkpoint_freq = checkpoint_freq
         if self._checkpoint_enable:
             assert self._checkpoint_freq is not None
-        self._stream_ignore_watermarks = set()  # input streams that do not send watermarks
+        # XXX(ionel): _checkpoints could be a queue, but we kept it a dict
+        # to be able to easily do asserts.
+        self._checkpoints = dict()
```

Comment:
It looks to me that old checkpoints are never garbage collected. IMHO, we need a method in op.py that garbage collects checkpoints up to a timestamp. I'll add in a separate commit.